test/test_quickbooks/test_quickbooks.py

```diff
@@ -0,0 +1,275 @@
+import unittest
+import requests_mock
+from parsons.etl.table import Table
+from parsons.quickbooks.quickbooks import QuickBooks
+from test_quickbooks_data import (
+    mock_groups_data,
+    mock_users_data,
+    mock_jobcodes_data,
+    mock_timesheets_data,
+    mock_schedule_calendars_list_data,
+    mock_schedule_events_data,
+    mock_geolocations_data,
+)
+
+
+class TestQuickBooks(unittest.TestCase):
+    @requests_mock.Mocker()
+    def setUp(self, mock_request):
+        self.qb = QuickBooks(token="abc123")
+        self.qb.url = "https://rest.tsheets.com/api/v1/"
+
+    def tearDown(self):
+        pass
+
+    @requests_mock.Mocker()
+    def test_qb_get_request(self, mock_request, end_point=None):
+        # Arrange
+
+        # If no end_point is passed, it will default to the groups endpoint
+        if end_point is None:
+            end_point = "groups"
+
+        querystring = {"page": 1}
+        mock_request.get(requests_mock.ANY, json=mock_groups_data)
+
+        # Act
+        result = self.qb.qb_get_request(end_point=end_point, querystring=querystring)
+
+        # Assert
+        self.assertIsInstance(result, Table)
+        self.assertIsInstance(end_point, str)
+        self.assertIsInstance(querystring, dict)
+
+    @requests_mock.Mocker()
+    def test_get_groups(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_groups_data)
+
+        # Act
+        result = self.qb.get_groups()
+
+        # Assert
+        self.assertIsInstance(result, Table)
```

Comment:
These tests only test whether the result is a Table, but you return an empty table for most of the methods. So you might also want to test the number of rows > 0 for all of these, and/or test the specific data being returned for each test.